### PR TITLE
FlexboxEngine clean up.

### PR DESCRIPTION
**Pull request by colinrtwhite**
Assorted clean up of the flexbox classes:
- Removes references to views.
- Create a `Direction` class to abstract away the flex direction and perform operations along the main/cross axis.
- Renames `Line` back to `FlexLine` since that's a better descriptor.
- Implement the PR feedback from https://github.com/cashapp/redwood/pull/372.

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Direction.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  if (this == FlexDirection.Row || this == FlexDirection.RowReverse) {
+    return Direction.Horizontal
+  } else {
+    return Direction.Vertical
+  }
+}
+
+/**
+ * An interface to perform operations along the main/cross axis without knowledge
+ * of the underlying [FlexDirection].
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Direction.kt:28]: NICE

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt [hunk 2]
```diff
@@ -379,18 +358,16 @@ public class FlexboxEngine {
...
         i >= toIndex && !reachedToIndex
       ) {
-        // Calculated to include a flex line which includes the flex item having the
-        // toIndex.
+        // Calculated to include a flex line which includes the flex item having the toIndex.
         // Let the sumCrossSize start from the negative value of the last flex line's
         // cross size because otherwise flex lines aren't calculated enough to fill the
         // visible area.
-        sumCrossSize = -flexLine.crossSize
+        sumCrossSize -= flexLine.crossSize
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt:365]: This is a deliberate behavior change?

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt [hunk 3]
```diff
@@ -1305,36 +1176,30 @@ public class FlexboxEngine {
...
       }
     }
   }
 
   /**
-   * Expand the view vertically to the size of the [crossSize] (considering the view margins)
-   *
-   * @param node the View to be stretched
-   * @param crossSize the cross size
-   * @param index the index of the view
```
Comment (swankjesse) [inline redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt:1332]: Thank you for deleting these

**Review by swankjesse — APPROVED** [review]

File: redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt (hunk 2 above)
Comment (colinrtwhite) [reply redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/FlexboxEngine.kt:365]: No! Good catch - that's a mistake.

**Pull request merged**